examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,414 @@
+different factors, including a long acquisition or high-resolution sampling.
+Currently, this example focuses on memory-efficient interactions with long
+time series fMRI data. In this case, loading the whole time series into memory
+may represent a significant computational cost. We will therefore explore
+strategies to minimize the amount of data that is loaded into memory,
+and we will compare these strategies against a naive usage of
+:class:`~nilearn.maskers.NiftiMasker`.
+
+To make this more concrete, we will create a large fMRI image with over
+800 time points by concatenating individual subjects in the
+:func:`~nilearn.datasets.fetch_development_fmri` dataset.
+Our goal is to extract data from several regions of interest (ROIs)
+defined by a number of binary masks, all in parallel.
+
+When using :class:`~nilearn.maskers.NiftiMasker` to extract data from each
+ROI in parallel, each parallel process will load the entire fMRI image into
+memory. This can lead to a significant increase in memory usage and may be
+infeasible in some computational environments.
+
+We will compare three different methods to mask the data from the fMRI image:
+
+1. A naive, unoptimized usage of :class:`~nilearn.maskers.NiftiMasker`
+2. Using array proxies defined with :mod:`nibabel`
+3. Using :class:`multiprocessing.shared_memory.SharedMemory`
+
+"""
+
+# %%
+# Create a large fMRI image
+# -------------------------
+# Here we will create a "large" fMRI image by fetch 5 subjects'
+# fMRI images via the :func:`~nilearn.datasets.fetch_adhd`
+# function, concatenating them and then saving to a file.
+
+from pathlib import Path
+
+from nilearn.datasets import fetch_adhd
+from nilearn.image import concat_imgs
+
+N_SUBJECTS = 6
+N_REGIONS = 6
+
+fmri_data = fetch_adhd(n_subjects=N_SUBJECTS)
+fmri_img = concat_imgs(fmri_data.func)
+
+output_dir = Path.cwd() / "results" / "plot_mask_large_fmri"
+output_dir.mkdir(parents=True, exist_ok=True)
+print(f"Large fmri file will be saved to:\n{output_dir}")
+
+fmri_path = output_dir / "large_fmri.nii.gz"
+fmri_img.to_filename(fmri_path)
+
+# %%
+# Create a set of binary masks
+# ----------------------------
+# We will now create 6 binary masks from a brain atlas. Here we will use the
+# multiscale functional brain parcellations via the
+# :func:`~nilearn.datasets.fetch_atlas_basc_multiscale_2015` function.
+# We will fetch a 64-region version of this atlas and then create separate
+# binary masks for the first 6 regions.
+
+from nilearn.datasets import fetch_atlas_basc_multiscale_2015
+from nilearn.image import load_img, new_img_like, resample_to_img
+
+atlas_path = fetch_atlas_basc_multiscale_2015(resolution=64).maps
+
+atlas_img = load_img(atlas_path)
+
+# %%
+# In this case, the atlas and the fMRI image have different resolutions.
+# So we will resample the atlas to the fMRI image. It is important to do that
+# because only :class:`~nilearn.maskers.NiftiMasker` can handle the resampling
```

Comment:
This is an interesting point (deserves an issue ?) but maybe out of topic here ?

Reply:
Sorry I don't understand what you are referring to? 

We do resampling beforehand here because when masking with numpy, the image array and mask array need to have the same dimensions.

And also to make sure that NiftiMasker isn't doing anything more than masking -- so that the comparison is apples-to-apples.

Reply:
Sorry, I misread the paragraph.